core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/DataTypeBuilder.scala

```diff
@@ -0,0 +1,29 @@
+package com.databricks.labs.remorph.parsers.snowflake
+
+import com.databricks.labs.remorph.parsers.snowflake.SnowflakeParser.Data_typeContext
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+
+import scala.collection.JavaConverters._
+object DataTypeBuilder {
+
+  def buildDataType(ctx: Data_typeContext): ir.DataType = {
+    val sizeOpt = Option(ctx.data_type_size()).map(_.getText.toInt)
+    ctx match {
+      case c if c.int_alias != null => ir.IntegerType()
```

Comment:
Snowflake has this weird challenge with datatypes, so I have seen data loss by mapping them to IntegerType directly. We should default to Number(38,0)  for all and, post-process, tackle this.

https://docs.snowflake.com/en/sql-reference/data-types-numeric#int-integer-bigint-smallint-tinyint-byteint